packages/ripple-address-codec/src/xrp-codec.ts

```diff
@@ -2,30 +2,24 @@
  * Codec class
  */
 
+import { base58xrp, BytesCoder } from '@scure/base'
 import { sha256 } from '@xrplf/isomorphic/sha256'
-import baseCodec = require('base-x')
-import type { BaseConverter } from 'base-x'
 
 import { arrayEqual, concatArgs, ByteArray } from './utils'
 
 class Codec {
   private readonly _sha256: (bytes: ByteArray) => Uint8Array
-  private readonly _alphabet: string
-  private readonly _codec: BaseConverter
+  private readonly _codec: BytesCoder
 
-  public constructor(options: {
-    sha256: (bytes: ByteArray) => Uint8Array
-    alphabet: string
-  }) {
+  public constructor(options: { sha256: (bytes: ByteArray) => Uint8Array }) {
     this._sha256 = options.sha256
-    this._alphabet = options.alphabet
-    this._codec = baseCodec(this._alphabet)
```

Comment:
Note to make sure this gets a special note in the migration guide since this is a reduction in functionality for this class.